borg/repository.py

```diff
@@ -591,6 +602,28 @@ def get_fd(self, segment):
             self.fds[segment] = fd
             return fd
 
+    def _sync_and_close(self, fd):
+        fd.flush()
+        os.fdatasync(fd.fileno())
```

Comment:
uhoh, while searching this i found that OSX doesn't make the guarantee that fsync really waits until all data arrived on disk.